New package: Fminbnd v1.0.0

JuliaRegistrator opened the pull request:
- Registering package: Fminbnd
- Repository: https://github.com/kagalenko-m-b/Fminbnd.jl
- Created by: @kagalenko-m-b
- Version: v1.0.0
- Commit: 976e4f3a3383a67e92e52286508e5819f8960f62
- Reviewed by: @kagalenko-m-b
- Reference: https://github.com/kagalenko-m-b/Fminbnd.jl/commit/976e4f3a3383a67e92e52286508e5819f8960f62#commitcomment-149775975
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd630959486abca3a91d3db37ab54cb7b879c4328290adba325ea8b2e487a38ff30c4673f23150774c5a2482525f1c600bfd7dbf87b7891d63ef1d0ff847018ef192a263f78a667fb94d01f1eecf89a39c8f9095f77eaa0a92a129c25ec7e6302cd268093b78b77ff009fe08670b7ade8020c05e6fc6ae1b244c41db448737e3980726660244d9df80cf0ca416bceb90250d218bd1d2640bcda3ed216348c5d74c39faa0b53c8a81fedb4c68ec1eb3f21c3 -->

goerz commented:
This seems to be the followup from https://github.com/JuliaRegistries/General/pull/99531.

I'm not really clear on what the `bnd` part of the name should indicate.

goerz commented:
I'd also strongly recommend leaving the name of the _function_ exported by the package as `fmin`.

kagalenko-m-b commented:
> I might tentatively suggest `NetlibFmin` as a name

There’re a widely used matlab/octave functions ‘fminbnd’ and python’s ‘fminbound’ that also derive from netlib’s fmin. “Bnd” means that it searches minimum on a bounded interval.

goerz commented:
Hmm… okay, but you're very explicitly translating the Netlib code, which doesn't use that name 🤷

goerz commented:
I would also point out that your signature does not match the [Matlab `fminbnd` function](https://www.mathworks.com/help/matlab/ref/fminbnd.html) (wrong order of arguments), but it does match the signature of Netlib's `fmin`

jaakkor2 commented:
Same Brent's method seems to be available in Optim.jl, see https://julianlsolvers.github.io/Optim.jl/stable/user/minimization/#Minimizing-a-univariate-function-on-a-bounded-interval

kagalenko-m-b commented:
> Same Brent's method seems to be available in Optim.jl, see https://julianlsolvers.github.io/Optim.jl/stable/user/minimization/#Minimizing-a-univariate-function-on-a-bounded-interval

Optim is a very large and complex package with many dependencies.

adienes commented:
rule 8 of package naming guidelines:

> 8. Avoid using a distinctive name that is already in use in a well known, unrelated project.

`fminbnd` is
 * distinctive
 * in use in MATLAB which is well known and
 * "unrelated" because you are not using its implementation


also rule 4

> 4. Err on the side of clarity, even if clarity seems long-winded to you.


I think `NetlibFmin` seems much more clear to me what this package does given your testimony that this is a 1-1 rewrite of that function.


naming aside, and I'm sorry to be too critical, but I would be a bit hesitant to say that this meets the quality standards to register in General in the first place. This package consists of a single function that is completely illegible. Could you at least try to factor out some of the `@goto`s into traditional control flow? compare to for example [scipy's implementation](https://github.com/scipy/scipy/blob/v1.14.1/scipy/optimize/_optimize.py#L2552-L2629)

goerz commented:
I wouldn't come down quite as hard on this. I think if it implemented the same interface as Matlab and was documented as such (instead of a translation of the Netlib code), I'd be totally fine with `Fminbnd`.

Likewise, the way it is implemented now, `NetlibFmin` would be totally fine. I'm saying this coming from a Fortran background where `Netlib` is the go-to if you need some code that you don't want to write yourself (most of it being in the public domain). So I don't even object to a Julia package that translates an existing Netlib package, even if it borders on being trivially small in this case.

> Could you at least try to factor out some of the @gotos into traditional control flow? 

With the direct reference to Netlib, it's kinda cute, and I don't think I'd mind. This is really as close of a translation of the original code as possible. It would be interesting to also write this with modern (haha) control flows and benchmark if there is any difference at all in performance.

kagalenko-m-b commented:
> “unrelated" because you are not using its implementation

Different implementation of the same algorithm is “unrelated”?

kagalenko-m-b commented:
> think if it implemented the same interface as Matlab and was documented as such (instead of a translation of the Netlib code), I'd be totally fine with `Fminbnd`

https://github.com/kagalenko-m-b/Fminbnd.jl/commit/b52ae0c07af2413a1de3bc8901f20b534a2b1c65

goerz commented:
Okay, that seems good to me… Just make sure that you retrigger the registration to really make sure that the correct commit gets tagged. Especially since you're using `v1.0.0` for this

JuliaRegistrator committed:
New package: Fminbnd v1.0.0

UUID: b7bab496-830c-4b53-8aec-8b9a988c40b4
Repo: https://github.com/kagalenko-m-b/Fminbnd.jl.git
Tree: f806e59cf762df964f7e91c082523d8e61e4c4f3

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

goerz commented:
Let me know when this is good to go, and I’ll apply the override to merge it

kagalenko-m-b commented:
> Let me know when this is good to go, and I’ll apply the override to merge it

I believe it’s as ready as it’ll ever be. Not added the tag yet, but the comment from registrar states that it should be done automatically via tagbot after the package registration.

goerz commented:
Yes, do not create tags to create releases. Releases have to be registered _first_ and then tagged afterwards (preferably via the bot)

kagalenko-m-b commented:
Looks like tagbot triggered automatically, thanks.
